src/external_app/external_auth.ts

```diff
@@ -63,6 +63,13 @@ class ExternalAuth extends Auth {
   public async refreshAccessToken() {
     const callbackPayload = { callback: CALLBACK_SET_TOKEN };
 
+    const callbackPromise = new Promise<RefreshTokenResponse>(
+      (resolve, reject) => {
+        window[CALLBACK_SET_TOKEN] = (success, data) =>
+          success ? resolve(data) : reject(data);
+      }
+    );
+
```

Comment:
We should add an `await 0` here so that we sleep 1 microtask to get the promise to actually set it on the window object.